tests/test_cuda_targeted_insertion_mover.py

```diff
@@ -593,6 +623,96 @@ def test_tibd_exchange_deterministic_moves(radius, proposals_per_move, precision
 
     params = nb.params[conf_idxs]
 
+    cutoff = nb.potential.cutoff
+
+    klass = custom_ops.TIBDExchangeMove_f32
+    if precision == np.float64:
+        klass = custom_ops.TIBDExchangeMove_f64
+
+    iterations = rng.integers(2, 5)
+
+    # Reference that makes proposals_per_move proposals per move() call
+    bdem_a = klass(
+        N,
+        center_group,
+        group_idxs,
+        params,
+        DEFAULT_TEMP,
+        nb.potential.beta,
+        cutoff,
+        radius,
+        seed,
+        proposals_per_move,
+        1,
+    )
+
+    bdem_b = klass(
+        N,
+        center_group,
+        group_idxs,
+        params,
+        DEFAULT_TEMP,
+        nb.potential.beta,
+        cutoff,
+        radius,
+        seed,
+        proposals_per_move * iterations,
+        1,
+        batch_size=batch_size,
+    )
+
+    iterative_moved_coords = conf.copy()
+    for _ in range(iterations):
+        iterative_moved_coords, _ = bdem_a.move(iterative_moved_coords, box)
+        assert not np.all(conf == iterative_moved_coords)
+    batch_moved_coords, _ = bdem_b.move(conf, box)
+
+    assert bdem_a.n_accepted() > 0
+    assert bdem_a.n_proposed() == proposals_per_move * iterations
+    assert bdem_a.n_accepted() == bdem_b.n_accepted()
+    assert bdem_a.n_proposed() == bdem_b.n_proposed()
+
+    # Moves should be deterministic regardless the number of steps taken per move
+    np.testing.assert_array_equal(iterative_moved_coords, batch_moved_coords)
+
+
+@pytest.mark.parametrize("radius", [1.0])
+@pytest.mark.parametrize("proposals_per_move, batch_size", [(5, 1), (100, 1), (12, 5), (120, 100), (1000, 1000)])
+@pytest.mark.parametrize("precision", [np.float64, np.float32])
+@pytest.mark.parametrize("seed", [2023])
+def test_tibd_exchange_deterministic_moves(radius, proposals_per_move, batch_size, precision, seed):
+    """Given one water the exchange mover should accept every move and the results should be deterministic if the seed and proposals per move are the same.
```

Comment:
That was incorrect, was removed: https://github.com/proteneer/timemachine/pull/1283/commits/18d218cc085e7f55258e0d4eca3ce6019d92484b

The test comment is more precise now, but there is a high acceptance rate since the box is quite large.